### PR TITLE
Refactor confirm_builtin_call, remove partial if

### PR DESCRIPTION
**Pull request by osa1**
Pass callee expr to `confirm_builtin_call`. This removes a partial
pattern match in `confirm_builtin_call` and the `panic` in the `else`
branch. The diff is large because of indentation changes caused by
removing the if-let.

### CODE REVIEW

File: compiler/rustc_typeck/src/check/callee.rs [hunk 1]
```diff
@@ -299,110 +306,104 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                     }
                 }
 
-                if let hir::ExprKind::Call(callee, _) = call_expr.kind {
```
Comment (osa1) [inline compiler/rustc_typeck/src/check/callee.rs:302]: This is the removed if-let. The rest are just indentation changes.

File: compiler/rustc_typeck/src/check/callee.rs [hunk 2]
```diff
@@ -299,110 +306,104 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
...
-                        if let Some(label) = label {
-                            err.span_label(span, label);
                         }
+                    };
+                    if let Some(label) = label {
+                        err.span_label(span, label);
                     }
-                    err.emit();
-                } else {
-                    bug!("call_expr.kind should be an ExprKind::Call, got {:?}", call_expr.kind);
```
Comment (osa1) [inline compiler/rustc_typeck/src/check/callee.rs:404]: This is the removed panic.

**Comment by osa1** [comment]
I annotated the important parts to make this easier to review. The rest are just formatting.

**Review by Julian-Wollersberger — APPROVED** [review]

**Pull request merged**